src/main/java/picocli/CommandLine.java

```diff
@@ -3267,6 +3265,34 @@ static IDefaultValueProvider createDefaultValueProvider(IFactory factory, Class<
             catch (Exception ex) { throw new InitializationException("Could not instantiate " + cls + ": " + ex, ex); }
         }
     }
+
+    public interface IHelpFactory {
+        public Help createHelp(Object command);
+        public Help createHelp(CommandSpec commandSpec);
+        public Help createHelp(CommandSpec commandSpec, Help.Ansi ansi);
+        public Help createHelp(CommandSpec commandSpec, Help.ColorScheme colorScheme);
+    }
+
+    public static abstract class BaseHelpFactory implements IHelpFactory {
```

Comment:
I prefer to have as few classes and interfaces as possible. Just a DefaultHelpFactory is enough I think.